- Added Throwable Brick And Nether Brick.

0rc1nus opened the pull request:
- Throwable Bricks will now slow down when it hits a glass
- Added names for the glow fire and the projectiles. So you won't get weird names when players play with Jade
- Changed the fungus type for the shroomloin entity

0rc1nus committed:
- Added Throwable Brick And Nether Brick.
- Throwable Bricks will now slow down when it hits a glass
- Added names for the glow fire and the projectiles. So you won't get weird names when players play with Jade
- Changed the fungus type for the shroomloin entity

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/events/MiscEvents.java:
```diff
@@ -84,31 +92,31 @@ public void onBlockBreak(BlockEvent.BreakEvent event) {
                 }
 
-                if (((ShroomloinEntity) entity).getFungusType() == 5) {
+                if (((ShroomloinEntity) entity).getFungusType() == 4) {
```
This should've been in a different PR

SwanX1 reviewed (changes requested):
Add a config to disable this feature

Hellion-Games commented on src/main/java/com/nekomaster1000/infernalexp/events/MiscEvents.java:
```diff
@@ -214,6 +222,85 @@ public void onRightClickItem(PlayerInteractEvent.RightClickItem event) {
+        RayTraceResult rayTraceResult = event.getRayTraceResult();
+        if (entity instanceof ThrowableBrickEntity) {
+            if (rayTraceResult.getType() == RayTraceResult.Type.BLOCK) {
+                if (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get()
+                    || world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get()) {
+                    world.playSound(null, ((BlockRayTraceResult)rayTraceResult).getPos(), IESoundEvents.QUARTZ_GLASS_TYPE.getHitSound(), SoundCategory.BLOCKS, 1.0F, 1.0F);
+                }
+                if ((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof GlassBlock
+                ||  world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof PaneBlock)
+                    && !((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get())
+                    || (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get())
+                    )) {
+                    world.destroyBlock(((BlockRayTraceResult)rayTraceResult).getPos(), false);
+                    ((ThrowableBrickEntity) entity).shoot(entity.getMotion().x, entity.getMotion().y, entity.getMotion().z, 0.3F, 1);
+                } else {
+                    ItemEntity itemEntity = new ItemEntity(world, entity.getPosX(), entity.getPosY(), entity.getPosZ(), Items.BRICK.getDefaultInstance());
+                    world.addEntity(itemEntity);
+                    entity.remove();
+                }
+            }
+        }
+        if (entity instanceof ThrowableNetherBrickEntity) {
+            if (rayTraceResult.getType() == RayTraceResult.Type.BLOCK) {
+                if (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get()
+                    || world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get()) {
+                    world.playSound(null, ((BlockRayTraceResult)rayTraceResult).getPos(), IESoundEvents.QUARTZ_GLASS_TYPE.getHitSound(), SoundCategory.BLOCKS, 1.0F, 1.0F);
+                }
+                if ((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof GlassBlock
+                    ||  world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof PaneBlock)
+                    && !((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get())
+                    || (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get())
+                    )) {
+                    world.destroyBlock(((BlockRayTraceResult)rayTraceResult).getPos(), false);
+                    ((ThrowableNetherBrickEntity) entity).shoot(entity.getMotion().x, entity.getMotion().y, entity.getMotion().z, 0.1F, 1);
+                } else {
+                    ItemEntity itemEntity = new ItemEntity(world, entity.getPosX(), entity.getPosY(), entity.getPosZ(), Items.NETHER_BRICK.getDefaultInstance());
+                    world.addEntity(itemEntity);
+                    entity.remove();
+                }
+            }
```
I'm pretty sure, since these are our own entities, this code can be done within the ThrowBrickEntity classes rather than through subscribing to an event. It would be a much more efficient way to handle it if it's possible. I'd look through the ProjectileEntity methods and see if one exists that would work for this.

0rc1nus committed:
- Added configs for triggering the shroomloins when a block is broken
- Replaced the projectile impact event by adding a new method in the throwable bricks classes

SwanX1 reviewed (changes requested):
NO I MEANT CONFIGS FOR THE BRICKS

0rc1nus committed:
- Removed Configs for triggering shroomloins
- Added Configs for the throwable bricks

SwanX1 reviewed (approved):
only issues are formatting issues, other than that LGTM

Dariensg commented:
One last thing Swan missed, there were no lang entries added for the config that you added.

Dariensg reviewed (changes requested):
One last thing Swan missed, there were no lang entries added for the config that you added.

0rc1nus committed:
- Added lang entry for the throwable bricks config

Dariensg reviewed (approved):
LGTM

0rc1nus committed:
- Moved the throwable bricks config from Mob Interactions to Miscellaneous category for clarity.
- Removed Silder for the throwable bricks

0rc1nus committed:
- Added Compat textures for the shroomloins
- Added Extra Biomes for the shroomloins to spawn in if byg is installed
- Added easter egg skins for the shroomloins
